realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -282,6 +284,44 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread until new changes to the Realm are available or {@link #stopWaitForChange}
+     * is called from another thread.
+     *
+     * @return {@code true} If the Realm was updated to the latest version, {@code false} if it was
+     * cancelled by calling {@link #stopWaitForChange}.
+     * @throws IllegalStateException if attempting to wait within a transaction or a Looper thread.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        if (Looper.myLooper() != null) {
+            throw new IllegalStateException("Cannot wait for changes inside a Looper thread. Use RealmChangeListeners instead.");
+        }
+        boolean hasChanged = sharedGroupManager.getSharedGroup().waitForChange();
+        if (hasChanged) {
+            // Since this Realm instance has been waiting for change, advance realm & refresh realm.
+            sharedGroupManager.advanceRead();
+            handlerController.refreshSynchronousTableViews();
+        }
+        return hasChanged;
+    }
+
+    /**
+     * Makes {@link #waitForChange()} return {@code false} immediately. This method is threadsafe
+     * and should be called from another thread than the one that called {@link #waitForChange}.
+     * Calling {@code #stopWaitForChange()} on a Realm that is not waiting for changes has no effect.
+     */
+    public void stopWaitForChange() {
+        // Check if the Realm instance has been closed
+        if (sharedGroupManager == null || !sharedGroupManager.isOpen()) {
```

Comment:
I'm afraid I cannot use it. It checks if an instance is called from the same thread. If not, it throws ISE.